### DISCUSSION TITLE
Pre-packaged Binary

### DISCUSSION THREAD

**Discussion by iloveyou3869** [orbitalquark/textadept#377] [Ideas]
TextAdept is a really cool software which I can't try.
I use Linux and the requirement to use TextAdept is to compile from source. Compiling from source process differs depending on which platform, Debian, Fedora, Suse, BSD. The documentation isn't clear on how to compile.
I am running Linux Mint Debian Edition Cinnamon KDE. What are the odds that I install the dependencies and Cinnamon crashes?
When Linus Torvolds was asked which Linux he personally uses He stated "Ubuntu". Someone cried out, "Why not Debian?"
Linus Torvolds responded he doesn't use a Linux OS where he has to compile everthing from source and they still don't work. He wants things to just work so he can get on with his life and programming.

LiteXL a not yet mature text editor provides a binary package so as many people as possible can use the app.
https://github.com/lite-xl/lite-xl/releases/download/v2.1.1/LiteXL-v2.1.1-x86_64.AppImage

Please provide a binary, appimage, so I can try TextAdept.

**Comment by rgieseke** [comment]
Binaries are available here:

https://github.com/orbitalquark/textadept/releases

**Reply by iloveyou3869** [reply]
It isn't an appimage.
or snap.
or Flatpak.
Documentation isn't clear on how to get the binary to run for linux.
Documentation says,, "On Linux you should compile from source instead of trying to run the binary."

**Reply by orbitalquark** [reply]
Have you even tried running the provided binaries? The manual is very clear on how to obtain and run Textadept on all 3 platforms, including Linux: https://orbitalquark.github.io/textadept/manual.html#getting-started. Where did you find that quote? I cannot find it.

**Reply by mhwombat** [reply]
> It isn't an appimage. or snap. or Flatpak.

**I suspect you're confused because the zipfile or tarfile contains a lot of files _in addition to the binary_.** Those are files that TextAdept needs to run. You don't need to build anything.

If you follow the instructions at https://orbitalquark.github.io/textadept/manual.html#getting-started, it will tell you how to proceed. If you get stuck, please tell us what you've tried, and provide any error messages you're getting.

**Reply by iloveyou3869** [reply]
> Have you even tried running the provided binaries? The manual is very clear on how to obtain and run Textadept on all 3 platforms, including Linux: https://orbitalquark.github.io/textadept/manual.html#getting-started. Where did you find that quote? I cannot find it.

"providing a single binary that runs on all Linux systems proves challenging, since the versions of software installed vary widely from distribution to distribution...you will need to compile Textadept manually for your system"

**Reply by orbitalquark** [reply]
You are selectively quoting. The real quote is (emphasis is mine) "**If Textadept will not start on your machine**, you will need to [compile](file:///home/me/code/textadept/docs/manual.html#compiling) Textadept manually for your system". Try the prebuilt binaries first. If they do not work, then you will have to resort to compiling manually. What you said before is disingenuous.

**Reply by iloveyou3869** [reply]
Read in error?

"Textadept can be run as a portable application. This is useful if, for example, you want to install Textadept onto a flash drive and take it with you for use on multiple machines. Normally, all settings and user data is stored in ~/.textadept/ (a local user directory that varies, depending on the platform). However, this user directory can be controlled using the -u or --userhome command line argument. For example, invoking textadept.exe with the command line arguments -u userdata will read from and store settings and user data to a userdata/ directory located inside an installation of Textadept. You can create a Windows shortcut that passes these command line arguments to the Textadept executable and use that shortcut to run Textadept portably.

Textadept’s user interface has been translated into a few different languages. When the application starts, it attempts to auto-detect your language settings by reading from the $LANG environment variable. If Textadept cannot determine what language to use, or if it does not support your language, it falls back on English. You can manually set your locale by copying one of the locale configuration files from Textadept’s core/locales/ to your ~/.textadept/ directory and renaming it locale.conf. If you would like to translate Textadept into your language, please translate the English messages in core/locale.conf and send me (see README.md) the modified file for inclusion in a future release.

macOS Tip: by default, macOS does not allow GUI applications like Textadept.app to see shell environment variables like $PATH. (The terminal version is unaffected.) Consequently, any features that utilize programs contained in $PATH (e.g. the programs in /usr/bin/ or /usr/local/bin/) will not find those programs. In order to work around this, Textadept automatically invokes a user-created ~/.textadept/osx_env.sh file when the application starts. This script should export all of the environment variables you need Textadept to see. For example:

export PATH=$PATH

Linux Note: providing a single binary that runs on all Linux systems proves challenging, since the versions of software installed vary widely from distribution to distribution. If you get errors like:

    error while loading shared libraries: <lib>: cannot open shared object file: No such file or directory
    /<path>/libc.so.6: version 'GLIBC_<version>' not found

you will need to compile Textadept manually for your system, which is a very straightforward and easy process."

**Reply by orbitalquark** [reply]
Fair enough, there has been a slight change in language from 11.4 to 12.0, but the premise in the manual is still the same: try running the pre-compiled binaries first. If you get errors, then you'll need to compile manually.

**Reply by iloveyou3869** [reply]
In the "textadept_11.4.linux" folder, there is a file called "textadept".
Right click on this file, properties, shows it is an application, permission, already selected as executable, close.
double click "textadept" -> Unknown File type: The file "textadept" has no known programs associated with it. Use the Open with dialog to pick a program to open it with.

**Reply by orbitalquark** [reply]
That's on your Linux file manager. Try double-clicking on the *textadept.desktop* file and see if that works. Otherwise, running Textadept is going to depend on your specific desktop environment. Look at the "Running" section of the manual for tips.

**Comment by ghost** [comment]
Lite XL is a community oriented project, in other words, it strives to suit the needs of the community. TextAdept _is not_.

Be familiar yourself with answers like this:

> Sorry to disappoint. I wrote Textadept for myself and decided to share it with you all. I don't need this feature, so I didn't implement it. I gave users that want it the option to implement it themselves. If this isn't your thing, then you may want to consider a more full-featured editor.

https://github.com/orbitalquark/textadept/discussions/372#discussioncomment-5298491

**Comment by ghost** [comment]
Btw, build from source is so easy, especially when you are on Linux. Just follow the steps. Did you read README.md?

https://github.com/orbitalquark/textadept/blob/default/README.md

**Comment by ghost** [comment]
From his init.lua, I guess the developer is using Ubuntu:

https://github.com/orbitalquark/.textadept/blob/default/init.lua

Ubuntu and Debian are very similar. I think it should work straight forward for you given you followed the instructions on README.md.

**Comment by ghost** [comment]
@iloveyou3869 Flatpak, Snap, AppImage are package formats. They are kind of like PortableApps on Windows. They are a way to distribute binary releases but not the only mean to distribute a binary release.

**Reply by ghost** [reply]
I think you mistakenly think textadept_nightly.linux.tgz is a source archive. It's a binary release. Just extract it and you will see.

**Reply by ghost** [reply]
At first I think you are asking for a portable TextAdept (truly portable with all of the dependencies included). It turned out to be wrong. You are just confusing things up.

**Reply by iloveyou3869** [reply]
I have never had success compiling anything from source.
Dependencies are the biggest reason for failed compilations. Most authors don't put much consideratation in the build documentation. They provide minimal information and expect everyone else to utilize an intuitive knowledge to fill in what they left blank.
Knowing exactly which dependencies are required to build is crucial.
Saying depends on GTK3 is not the same thing as saying it depends on these GTK3 files specifically.
I am currently using a Debian based distro. Notorius for being a PITA, no Falafal.

I am attempting to learn programming, Go, Elixir, Lisp; Absolutely NO NO NO Javascript, Ever ;D.
I need a good quality text editor, my favorite is Sublime Text. I like ST because it functions well as a general purpose text editor that can be a full IDE if needed.

I think TextAdept would make a good framework for creating a general purpose Dr. Racket like development editor.

**Comment by iloveyou3869** [comment]
I fear having to reinstall my OS because I installed gnome packages on KDE Cinnamon. :sweat_drops: 
I succeeded in building the package from source.
In the Compile section of the manual (The last Compile section)
There is a list of dependancies, 3 without Docker.
Installed these 3 dependencies, crossed my fingers and make deps in src.
I didn't need to reinstall the OS. :sweat: 
No Command Pallet?
Simple.
In the toolbar, buffer>lexer shows a list of languages it can recognize.

**Comment by iloveyou3869** [comment]
Thank you for the guidance you gave.

**Comment by orbitalquark** [comment]
I am locking this discussion due to some less-than-civil comments directed at OP. I have removed those comments. Please be respectful.
